app/controllers/settings/dex_connector_oidcs_controller.rb

```diff
@@ -0,0 +1,94 @@
+# Settings::DexConnectorOidcsController is responsible to manage requests
+# related to OIDC connectors.
+class Settings::DexConnectorOidcsController < SettingsController
+  # populate @data_holder with existing database record
+  before_action :set_data_holder, except: [:index, :new, :create]
+
+  def index
+    @oidc_connectors = DexConnectorOidc.all
+  end
+
+  def new
+    @is_data_valid = false
+    @data_holder = data_holder_type.new
+  end
+
+  def edit
+    @is_data_valid = false
+  end
+
+  def create
+    @data_holder = data_holder_type.new(data_holder_params)
+
+    if params[:validate]
+      @is_data_valid = @data_holder.valid?
+      if @is_data_valid
+        flash.now[:notice] = "#{@data_holder.class} is valid"
```

Comment:
We can simplify with just `flash[:notice]`. 

Also the current message is `DexConnectorOidc is valid`, wouldn't it be better if we used `OIDC Connector is valid`? What do you think?

Reply:
If we just use `flash`, then if the user doesn't clear the alert, it persists. So, enter an invalid value, validate, then enter a valid value. On that page, there will be both the "invalid" warning and the "valid" alert. Using `flash.now` mitigates that issue.

The message should probably be changed; I'll do that when I get back. :)

Reply:
You are right about `now`. Never had to use it though.

Reply:
I've only known about it for a few hours. ;)

Better friendly_name implemented; will push after re-running the test cases.

Reply:
ad72e37